test/k8sT/Services.go

```diff
@@ -1548,6 +1549,155 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`, helpers.DualStackSupp
 		})
 	})
 
+	SkipContextIf(func() bool {
+		// The graceful termination feature depends on enabling an alpha feature
+		// EndpointSliceTerminatingCondition in Kubernetes.
+		return helpers.RunsOnGKE() || helpers.RunsOnEKS()
+	}, "Checks graceful termination of service endpoints", func() {
+		const (
+			clientPodLabel = "app=graceful-term-client"
+			serverPodLabel = "app=graceful-term-server"
+			testPodLabel   = "zgroup=testDSClient"
+		)
+		var (
+			gracefulTermYAML string
+			clientPod        string
+			serverPod        string
+			wg               sync.WaitGroup
+		)
+
+		waitForServer := func() {
+			err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l app=graceful-term-server", helpers.HelperTimeout)
+			ExpectWithOffset(1, err).Should(BeNil(), "Server pods failed to come up")
+			pods, err := kubectl.GetPodNames(helpers.DefaultNamespace, serverPodLabel)
+			ExpectWithOffset(1, err).Should(BeNil(), "Cannot retrieve pod names by filter %s", serverPodLabel)
+			serverPod = pods[0]
+			podIPs, err := kubectl.GetPodsIPs(helpers.DefaultNamespace, serverPodLabel)
+			ExpectWithOffset(1, err).Should(BeNil(), "Cannot retrieve pod IPs for %s", serverPodLabel)
+			ExpectWithOffset(1, len(podIPs)).To(Equal(1), "Unexpected number of service backends")
+			for _, ip := range podIPs {
+				err = kubectl.WaitForServiceBackend(helpers.K8s1, ip)
+				ExpectWithOffset(1, err).Should(BeNil(), "Failed waiting for %s backend entry on k8s1", ip)
+				err = kubectl.WaitForServiceBackend(helpers.K8s2, ip)
+				ExpectWithOffset(1, err).Should(BeNil(), "Failed waiting for %s backend entry on k8s2", ip)
+			}
+		}
+
+		terminateServiceEndpoint := func() {
+			By("Deleting service endpoint pod %s", serverPodLabel)
+			wg.Add(1)
+			// Delete the service pod asynchronously subsequent steps need
+			// to be checked while the pod is terminating.
+			go func() {
+				defer wg.Done()
+				res := kubectl.DeleteResource("pod", fmt.Sprintf("-n %s -l %s", helpers.DefaultNamespace, serverPodLabel))
+				ExpectWithOffset(1, res).Should(helpers.CMDSuccess(), "Unable to delete %s pod", serverPodLabel)
+			}()
+
+			By("Waiting until server is terminating")
+			ctx, cancel := context.WithCancel(context.Background())
+			res := kubectl.LogsStream(helpers.DefaultNamespace, serverPod, ctx)
+			Expect(res.WaitUntilMatch("terminating")).To(BeNil(),
+				"%s is not in the output after timeout", res.GetStdOut())
+			defer func() {
+				cancel()
+				res.WaitUntilFinish()
+			}()
+		}
+
+		BeforeAll(func() {
+			if helpers.SkipK8sVersions("<1.20.0") {
```

Comment:
(GH doesn't allow me to comment in my previous comment) I'd still move it to the ctx skip condition. You can add the comment there saying why we skip the test. Otherwise, it makes it more difficult already overbloated and poorly structured suite file (I'm talking about the whole `K8sService` suite).